### DISCUSSION TITLE
Plot pole figures for the simulated Euler angles

### DISCUSSION THREAD

**Discussion by mohammadsattari1990** [mtex-toolbox/mtex#2062] [Ask Anything]
Hi all,
I have simulated the microstructure evolution of 316L and the "angles.txt" file contains the simulated Euler angles corresponding to a cross-section. These Euler angles are defined to rotate the specimen coordinates (XYZ) to the crystal coordinates. The corresponding experimental pole figures can be seen in a photo named "Experimental PoleFigures". I am using MTEX and the following script to plot pole figures for these simulated Euler angles (in the "angles.txt" file). This script creates 2 figures, for the first one, the Euler angles have been used directly to create an orientation object. Since in MTEX, the Euler angles rotate the crystal coordinates to the specimen coordinates, the 2nd figure belongs to the Euler angles corresponding to the inverse of the 1st orientation object. I expect that the method to create the second figure is wrong BTW.

```matlab
% Settings
plotx2east
plotzOutOfPlane
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

% Euler angles, symmetry and Miller indices
angles = readmatrix('angles.txt');
CS = crystalSymmetry('Fm-3m');
%SS = specimenSymmetry('triclinic');
h = Miller({1,0,0}, {0,1,1}, {1,1,1}, CS);

% Orientations and pole figures
ori = orientation.byEuler(angles(:,1), angles(:,2), angles(:,3), CS);
odf = calcDensity(ori);
figure
plotPDF(odf, h)

% Inverse of the orientations and corresponding pole figures!!!
ori_inv = inv(ori);
angles_inv = [ori_inv.phi1, ori_inv.Phi, ori_inv.phi2];
ori2 = orientation.byEuler(angles_inv(:,1), angles_inv(:,2), angles_inv(:,3), CS);
odf_inv = calcDensity(ori2);
figure
plotPDF(odf_inv, h)
```

The problem is that the pole figures for the simulated Euler angles (see below) do not match the experimental ones. To check if there are any issues with the Euler angles, I used the "plotPoleFigure" function (attached, and its format needs to be changed to .m), and its results (labeled as "Using plotPoleFigure function") match the experimental ones. So there should be some remarks when using MTEX for such purposes. I appreciate it if you could assist me with this issue and provide me with a script that can correctly plot the pole figures for the simulated Euler angles.

**Simulated Euler angles:**
[angles.txt](https://github.com/mtex-toolbox/mtex/files/14172185/angles.txt)

**Experimental Pole Figures:**
![Experimental PoleFigures](https://github.com/mtex-toolbox/mtex/assets/33626759/994dbd7f-7d4b-4a70-a141-ff54fb80cfa4)

**Pole figures corresponding to the simulated Euler angles (in the "angles.txt" file), using the script above:**
![Simulation](https://github.com/mtex-toolbox/mtex/assets/33626759/80441d64-e557-4d5b-8666-b4dcf9a28fbe)

**plotPoleFigure function:**
[plotPoleFigure.txt](https://github.com/mtex-toolbox/mtex/files/14172191/plotPoleFigure.txt)

**Using plotPoleFigure function:**
![Using plotPoleFigure function](https://github.com/mtex-toolbox/mtex/assets/33626759/47ba7d9b-cd3b-4db1-990c-4bde0e29f466)

**Comment by ralfHielscher** [comment]
In you question its unclear which figure corresponds to which command.

**Reply by mohammadsattari1990** [reply]
I edit the thread to make it more clear. Thank you!

**Comment by ralfHielscher** [comment]
Hi Mohammad,
the only difference between your function `plotPoleFigure` and the MTEX code is that your function treats the Euler angles in degree whereas your above code treats them in radians. I had a look at the data and would say the latter is correct. Hence, I would go with the second line of pole figures in your post.
Ralf.

**Reply by mohammadsattari1990** [reply]
Hi Ralf,

Thank you for your prompt answer.
I see! the ```plotPoleFigure``` and MTEX create the same results if the angle unit is consistent. So in this case, my simulation results do not match the experiment and there should be issues in my Euler angles.

Mohammad
